src/node_crypto.cc

```diff
@@ -5503,14 +5503,21 @@ void RandomBytesCheck(RandomBytesRequest* req, Local<Value> argv[2]) {
       ERR_error_string_n(req->error(), errmsg, sizeof errmsg);
 
     argv[0] = Exception::Error(OneByteString(req->env()->isolate(), errmsg));
-    argv[1] = Null(req->env()->isolate());
     req->release();
   } else {
     char* data = nullptr;
     size_t size;
     req->return_memory(&data, &size);
     argv[0] = Null(req->env()->isolate());
-    argv[1] = Buffer::New(req->env(), data, size).ToLocalChecked();
+
+    if (Buffer::HasInstance(argv[1])) {
+      char* buf = reinterpret_cast<char*>(Buffer::Data(argv[1].As<Object>()));
+      for (size_t i = 0; i < size; i++) {
+        buf[i] = data[i];
+      }
```

Comment:
Would there be any reason not to fill the passed buffer directly instead of copying?